feat: Translate docs 2.0 (#868)
ZupIT/beagle-docs: content/en/android/customization/view-client.md

---

## What is it?

The `View Client` is very similar to the HttpClient. While the HttpClient is responsible for handling requests (views, json data, images, etc), the ``ViewClient`` is only responsible for fetching views (Server Driven Views).

> The ViewClient interface has to functions **`fetch`** and **`prefetch`**.

Reviewer: ```suggestion
The ViewClient interface has two functions **`fetch`** and **`prefetch`**.
```

The **`fetch`** default implementation has 2 functionalities:

> 1. It searches for a response data in a local cache, and if a cache exists, this function removes it from cache and return its contents;

Reviewer: Is it removed from the cache or only retrieved?

Author: The previous description says it is removed but in a different way.

> 2. If there is no response data in cache, this function calls the `HttpClient` and make a request for a ResponseData.

Reviewer: ```suggestion
2. If there is no response data in cache, this function calls the `HttpClient` and makes a request which returns a ResponseData.
```

> The **`prefech`** default implementation has 2 functionalities:

Reviewer: ```suggestion
The **`prefetch`** default implementation has 2 functionalities:
```

1. It searches for a response data in a local cache, and it returns this cache if it exists;

> 2. If there is no response data in cache, this function calls the `HttpClient` and make the request for a ResponseData and store it in cache (memory).

Reviewer: Both fetch and preFetch seem to be very similar in definition. Is that so? Do both of them work with the cache storage?

Author: Yes. Both could handle cache

```

> Above we have implemented a logic to `fetch` and `prefetch` that will store every fetch result in memory using a MutableMap called `cachedResponses`. It uses the *request url* as key, and search for a ``ResponseData`` in it every time `fetch` or `prefetch` is called. This is a simple implementation, since this cache would never expire, and the objective here is to show how this feature could be implemented using the ViewClient.

Reviewer: ```suggestion
Above we have implemented a logic to `fetch` and `prefetch` that will store every fetch result in memory using a MutableMap called `cachedResponses`. It uses the *request url* as key, and searches for a ``ResponseData`` in it every time `fetch` or `prefetch` is called. This is a simple implementation, since this cache would never expire, and the objective here is to show how this feature could be implemented using the ViewClient.
```
